### ISSUE TITLE
Unable to Install Addons in Nvda 2020.2 Beta 2

### ISSUE THREAD

**Issue by akj** [nvaccess/nvda#11347]
### Steps to reproduce:
1. Download an addon, for this example let's use [Kill NVDA](https://addons.nvda-project.org/addons/killNVDA.en.html) though I have tested with many others.
2. Click on the .nvda file to trigger install.
3. Sad beep, no install.

#### NVDA Log Output
```
ERROR - RPC process 2156 (nvda_slave.exe) (11:45:11.579) - Dummy-6 (8036):
__main__.main:
slave error
Traceback (most recent call last):
  File "nvda_slave.pyw", line 100, in main
ValueError: No such action
```
### Expected behavior:
The addon is installed.
#### NVDA installed/portable/running from source:
Installed
#### NVDA version:
2020.2 Beta 2
#### Windows version:
Windows 10 Pro 1909, Build 18363.90
#### Does the issue still occur after restarting your computer?
Yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
No
#### If addons are disabled, is your problem still occuring?
Yes
#### Did you try to run the COM registry fixing tool in NVDA menu / tools?
No

**Comment by DrSooom** [comment]
@AKJ: You mean [NVDA 2020.2 Beta 2](https://ci.appveyor.com/api/buildjobs/apcqhdny4fhby273/artifacts/output/nvda_2020.2beta2.exe), correct? Because 2020.1 Beta 2 doesn't exist.

**Comment by AKJ** [comment]
Yes - I blame a lack of coffee. Latest beta release.

**Comment by leonardder** [comment]
This error is pretty weird. What language of Windows are you using? It could be helpful if you could try NVDA 2020.1 to see whether the issue also exists there. Could you also try to install the add-on from the add-ons manager?

**Comment by AKJ** [comment]
Okay, tried a few more things. The tl;dr is that I think this may be a problem with my installation or version of Firefox (78.0.2). Running NVDA in English.

- Choosing "open" after downloading from Firefox triggers the error.
- Activating the addon from the "library" view (accesible with CTRL+J) triggers the error.
- Activating the addon directly from the %tmp% folder does not trigger the error.
- Installing the addon from the addons manager does not trigger the error.

Hope this helps. I suspect this issue can be closed given this new information, but if you have any other thoughts or things I should test I'm happy to do so.

**Comment by josephsl** [comment]
Hi, can you tell us what add-on you were trying to install? That way we can figure out if it is indeed FF or something going on with add-on file host. Thanks.

**Comment by AKJ** [comment]
I tried with several addons from the [NVDA addons](https://addons.nvda-project.org/) page including [Kill NVDA](https://addons.nvda-project.org/addons/killNVDA.en.html), [Phonetic Punctuation](https://addons.nvda-project.org/addons/phoneticPunctuation.en.html), and even [Windows 10 App Essentials]9https://addons.nvda-project.org/addons/wintenApps.en.html).

**Comment by josephsl** [comment]
Hi, I see. If you haven’t, can you try downloading and installing these add-ons with another browser, say Chrome? Thanks.

**Comment by AKJ** [comment]
Can confirm that this issue only occurs in Firefox. I only tested with 78.0.2.

**Comment by DrSooom** [comment]
@AKJ: Could you please try to reproduce this with [Firefox Portable](https://portableapps.com/apps/internet/firefox_portable) as well? Thanks.

**Comment by Adriani90** [comment]
@AKJ I think this might be a certificate issue on your side. Could you please try to re install Firefox, or to clean your firefox cache?

**Comment by lukaszgo1** [comment]
I don't think so - it is also reproducible for me.

**Comment by Adriani90** [comment]
@lukaszgo1 since you are still able to reproduce this, did you try to reinstall Firefox on your machine and see if it is still occuring?

**Comment by lukaszgo1** [comment]
Yes, I did. Reinstall of Firefox hasn't fixed the problem, but creation of the clear profile did the trick. Since I'm not very font off recreating my Firefox config from scratch I'll investigate this further starting with adding additional logging to NVDA_Slave. Don't close this just yet!

**Comment by DrSooom** [comment]
@lukaszgo1: Or just create a new profile in Firefox first. Enter "about:profiles" in the address bar.

@Adriani90: See [this German Firefox support article regarding uninstalling Firefox and removing Firefox profiles](https://support.mozilla.org/de/kb/Firefox-deinstallieren).

**Comment by seanbudd** [comment]
Closing this until it can be reproduced again

**Comment by lukaszgo1** [comment]
Just FYI @seanbudd This is still reproducible for me.

**Comment by Adriani90** [comment]
cc:_ @jcsteh since it is related to Firefox only.

**Comment by jcsteh** [comment]
I can't reproduce this. @lukaszgo1, can you still reproduce it a year later? I'm not quite sure how to suggest diagnosing this yet, but I guess a good starting point would be trying to get the command line passed to nvda_slave when this fails. That will require a build with some debugging code, though, so I'd like to confirm it's actually still a problem before suggesting further diagnosis.

**Comment by lukaszgo1** [comment]
Yes, this is still reproducible for me using latest Firefox Nightly on a different Windows install.
To debug this I've replaced `nvda_slave` with a simple command line program which just prints its CLI arguments. Normally when opening an add-on the first argument should be `addons_installAddonPackage`, and the second the path to the add-on file, however when the add-on is opened from Firefox the `addons_installAddonPackage` is missing, and only add-on path is shown. Looking at the Firefox's action for downloaded files (settings -> applications group) I see there are two possible actions for nvda-addon files, the first is 'Use NVDA (default)', and the second 'Use NVDA'. When the first one- that is 'Use NVDA (default)' is selected add-ons can be installed normally. With the second one selected (which was the default until I've changed nvda-addon files to open in 7-zip to work around this bug_ the error presented in the issue description is shown, and only add-on path is provided to `nvda_slave`. By the way it would be immensely useful if Firefox's list of file types supported type ahead - scrolling with arrows and page up/ down is counter productive with so many entries.

**Comment by Adriani90** [comment]
cc: @seanbudd for priority triaging.

**Comment by seanbudd** [comment]
@lukaszgo1 - can you confirm this is still an issue?

**Comment by SaschaCowley** [comment]
I was able to reproduce this with the details provided in https://github.com/nvaccess/nvda/issues/11347#issuecomment-1493308289

**Comment by Adriani90** [comment]
@jcsteh was there any update in Firefox that touched on this issue described in https://github.com/nvaccess/nvda/issues/11347#issuecomment-1493308289?

Is this issue related to NVDA at all?

**Comment by jcsteh** [comment]
I still can't reproduce this. I also don't have an entry in Firefox Settings -> Applications for NVDA add-ons. I can't reproduce it with a clean profile either.

The fact that two people can reproduce this suggests this is not entirely system specific. However, I'm wondering whether this might have been something that somehow ended up in configurations a while ago, but has since been fixed.

Can either of you (@lukaszgo1, @SaschaCowley) reproduce this with a clean profile? You can create a new profile by running firefox with the -profilemanager command line argument.

**Comment by seanbudd** [comment]
Can anyone still reproduce this? perhaps it was inadvertently fixed in Firefox?

**Comment by seanbudd** [comment]
Closing as fixed
